primitiveFTPd/src/org/primftpd/filesystem/StorageManagerUtil.java

```diff
@@ -47,6 +54,44 @@ public static String getFullDocIdPathFromTreeUri(@Nullable final Uri treeUri, Co
         }
     }
 
+    public static int getFilesystemTimeResolutionForTreeUri(Uri startUrl) {
+        logger.trace("getFilesystemTimeResolutionForTreeUri({})", startUrl);
+        String mountPoint = "/mnt/media_rw/" + getVolumeIdFromTreeUri(startUrl);
+        try(BufferedReader br = new BufferedReader(new FileReader("/proc/mounts"))) {
+            // sample contents for /proc/mounts
+            // /dev/block/vold/public:xxx,xx /mnt/media_rw/XXXX-XXXX vfat ... 0 0                     -> 2000 ms
+            // /dev/block/vold/public:xxx,xx /mnt/media_rw/XXXX-XXXX sdfat ...,fs=vfat:16,... 0 0     -> 2000 ms
+            // /dev/block/vold/public:xxx,xx /mnt/media_rw/XXXX-XXXX sdfat ...,fs=vfat:32,... 0 0     -> 2000 ms
+            // /dev/block/vold/public:xxx,xx /mnt/media_rw/XXXX-XXXX sdfat ...,fs=exfat,... 0 0       ->   10 ms
+            for (String line; (line = br.readLine()) != null; ) {
```

Comment:
log "line" as trace, please, that will help if some users report issues with this somewhen in future